bin/spark-class

```diff
@@ -68,15 +68,26 @@ fi
 # The exit code of the launcher is appended to the output, so the parent shell removes it from the
 # command array and checks the value to see if the launcher succeeded.
 build_command() {
-  "$RUNNER" -Xmx128m -cp "$LAUNCH_CLASSPATH" org.apache.spark.launcher.Main "$@"
+  "$RUNNER" -Xmx128m $SPARK_LAUNCHER_OPTS -cp "$LAUNCH_CLASSPATH" org.apache.spark.launcher.Main "$@"
   printf "%d\0" $?
 }
 
 # Turn off posix mode since it does not allow process substitution
 set +o posix
 CMD=()
-while IFS= read -d '' -r ARG; do
-  CMD+=("$ARG")
+DELIM=$'\n'
+while IFS= read -d "$DELIM" -r ARG; do
+  if [ -n "$CMD_START_FLAG" ]; then
```

Comment:
Better to initialize `CMD_START_FLAG` to an empty string before the loop.

Or you could use `DELIM` for the same effect.

Reply:
Thanks. I'll initialize the flag.